packages/server/api/src/app/ai/config/ai-config.controller.ts

```diff
@@ -15,7 +16,26 @@ export const aiConfigController: FastifyPluginAsyncTypebox = async (app) => {
     '/',
     SaveAiConfigOptions,
     async (request, reply): Promise<AiConfig> => {
-      const { valid, error } = await validateAiProviderConfig(request.body);
+      let existingApiKey = request.body.apiKey;
+      if (request.body.id) {
+        const exisiting = await aiConfigService.get(
+          {
+            projectId: request.principal.projectId,
+            id: request.body.id,
+          },
+          false,
+        );
+
+        existingApiKey = exisiting
+          ? encryptUtils.decryptString(JSON.parse(exisiting.apiKey))
```

Comment:
why do we need a JSON parse?

Reply:
because its an encrypted object, not a string

Reply:
I don't understand, the method is called decryptString.
Isn't it `encryptUtils.decryptString(exisiting.apiKey)`?

Reply:
ew as we discussed, it's "decryptToAString"